src/gsi/unfill_mass_grid2.f90

```diff
@@ -381,3 +381,107 @@ subroutine unfill_mass_grid2t_drycheck(gout,nx,ny,gin,qs)
   end do
   
 end subroutine unfill_mass_grid2t_drycheck
+
+subroutine unfill_mass_grid2t_ldlkmask(gout,nx,ny,gin,landmask, &
+                            lakemask,i_howv_mask)
+!$$$  subprogram documentation block
+!                .      .    .                                       .
+! subprogram:    unfill_mass_grid2t        opposite of fill_mass_grid2
+!   prgmmr: parrish          org: np22                date: 2004-06-22
+!
+! abstract: This is the same as unfill_mass_grid2t and unfill_mass_grid2t_ldmk
+!           but screen off analysis increment over land and/or lake . 
+!
+!
+! program history log:
+!   2004-07-16  parrish
+!   2013-10-25  todling - reposition ltosi and others to commvars
+!   2014-03-12  Hu       Code for GSI analysis on grid larger than background grid: 
+!                        Here input grid is larger than output grid.
+!   2014-04-04  todling - reposition ltosi and others to commvars
+!   2015-01-15  Hu      - apply the land/sea mask here for soil adjustment
+!                          fields
+!   2025-01-28  zhao    - apply land and lake mask for howv (wave height) fields
+!
+!   input argument list:
+!     gout     - input A-grid (reorganized for distibution to local domains)
+!     gin      - preexisting input values to be added to on C-grid
+!     nx,ny    - input grid dimensions
+!     landmask - land mask (0: water, 1: land)
+!     lakemask - lake mask (0: sea and land (including Great Lakes), 1: lake)
+!     i_howv_mask - input option for HOWV adjustment
+!                  = 0: no adjustment
+!                  = 1: mask off data over land
+!                  = 2: mask off data over land and lake
+!
+!   output argument list:
+!     gin      - output result on C grid
+!
+! attributes:
+!   language: f90
+!   machine:  ibm RS/6000 SP
+!
+!$$$
+  use kinds, only: r_single,i_kind
+  use gridmod, only: itotsub,iglobal
+  use general_commvars_mod, only: ltosi,ltosj
+  use mod_wrfmass_to_a, only: wrfmass_a_to_h4
+  use gridmod, only: nlon, nlat
+
+  implicit none
+
+  integer(i_kind), intent(in   ) :: nx,ny
+  real(r_single) , intent(in   ) :: gout(itotsub)
+  real(r_single) , intent(inout) :: gin(nx,ny)
+  real(r_single) , intent(in)    :: landmask(nx,ny)
+  real(r_single) , intent(in)    :: lakemask(nx,ny)
+  integer(i_kind), intent(in   ) :: i_howv_mask
+  
+  real(r_single) ba(nlon,nlat)
+  real(r_single) b(nx,ny)
+  integer(i_kind) i,j
+
+  do i=1,iglobal
+     ba(ltosj(i),ltosi(i))=gout(i)
+  end do
+
+  if(nlon == nx .and. nlat == ny) then
+     b=ba
+  else
+     call wrfmass_a_to_h4(ba,b)
+  endif
+
+! Only add analysis increment over sea water (no land and lake)
+  select case (i_howv_mask)
+    case(2)
+      write(6,*) 'unfill_mass_grid2t_ldlkmask: using landlakemask to screen off HOWV analysis increment over land and lake.'
```

Comment:
Do multiple mpi tasks call this routine?  If so, `write(6,*)` could generate a lot of output.  Do we need this output from all tasks?  Should we place write statements in this routine inside verbose blocks?

Reply:
@RussTreadon-NOAA 
Hi Russ,

I double checked the code, in GSI this subroutine is only invoked on **pe 0** . So it should not be called by multiple MPI tasks simultaneously.
in subroutine **wrwrfmassa_netcdf_wrf**(this,mype)
`...`
`      if(mype == 0) then` 
`...`
`          if ( i_howv_mask > 0 ) then`
`             call unfill_mass_grid2t_ldlkmask(tempa,im,jm,temp1,landmask, &`                                         
`                                                            lakemask,i_howv_mask)`
`...`
`   end if     !endif mype==0`
`...`

Thank you!